docker/dev-builder/ubuntu/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
```

Comment:
Do we need to upgrade the base image for dev-builder? The change may upgrade the glibc version.

Reply:
I think the dev-build base image should be upgrade first by: https://github.com/GreptimeTeam/greptimedb/pull/5554

use the `ubuntu-22.04` image to build the `dev-build` image, and then use the latest `dev-build` image tag to modify here: https://github.com/GreptimeTeam/greptimedb/blob/main/Makefile#L11

Reply:
20.04 is reaching its end of life in May. The apt repository will be offline at the time.

Reply:
Release `dev-build` image in: https://github.com/GreptimeTeam/greptimedb/actions/runs/13646694372